torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -95,9 +117,10 @@ void setstateTuple(const IValue& ivalue, std::vector<c10::IValue>& elements) {
   auto type = obj->type();
   if (checkHasValidSetGetState(type)) {
     Function *setstate = type->getMethod("__setstate__");
-    elements.push_back(getFunctionTuple(*setstate));
-  }
-  else {
+    if (setstate->isGraphFunction()) {
+      elements.push_back(getFunctionTuple(*setstate));
```

Comment:
To my understanding, it supports both graphFunction and non-graphFunction? That would be nice! Could you also cover graphFunction in unit test?

Reply:
__setstate__ with a non-Graph function is handled specially in the load path. This change is basically just making it so we don't try to serialize those (serialization wouldn't make sense since we can't get the Function's graph)